eZ/Publish/SPI/Persistence/User/Policy.php

```diff
@@ -81,4 +81,25 @@ class Policy extends ValueObject
      *                   Can not be a empty array as '*' should be used in this case.
      */
     public $limitations;
+
+    /**
+     * Return true if at least one of the given policies overlaps this policy (has a wider scope).
+     *
+     * @param \eZ\Publish\SPI\Persistence\User\Policy[] $policies
+     * @return bool
+     */
+    public function overlaps(array $policies)
```

Comment:
Move to be private method on `Repository` _(actually, please rebase to 6.6, as we are about to tag last tag og 6.5, so that will be PermissionResolver)_. And while at it change to something like `private function isOverlappedByWiderPolicy(spiPolicy $policy, array $policies)`